src/components/SingleArticlePage.jsx

```diff
@@ -29,43 +28,83 @@ const fetchArticle = () => {
 const fetchComments = () => {
   return fetch(`https://nc-news-z0zw.onrender.com/api/articles/${articleid}/comments`)
   }
-const upVote = () => {
+const changeVoteArticle = (value) => {
+  return fetch(`https://nc-news-z0zw.onrender.com/api/articles/${articleid}`, {
+    method: 'PATCH',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({ inc_votes: value}),
+  })
+}
 
+const upVoteArticle = (event) => {
+  if (vote ===  1){
+    console.log("already upvoted")
+    return
+  }
+  event.preventDefault()
+  changeVoteArticle(1).then(({status}) => status === 200 ? setVote((currentVote)=> currentVote + 1) : setVote("unavailable"))
```

Comment:
I think this has been achieved now by just setting the vote state prior to the api call, if the api call fails it will show voting unavailable still.  Maybe I need to add something like 'voting is unavailable, your vote has not been counted at this time'